### PR TITLE
[tests-only] [full-ci] Added spaces for suite `apiWebdavOperations`

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
 Added Spaces for suite apiWebdavOperations 

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/3049

## How Has This Been Tested?

- CI
- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIUpload-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34452/151/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavOperations-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34456/94/1

File: tests/acceptance/features/apiWebdavOperations/deleteFolderContents.feature [hunk 1]
```diff
@@ -24,9 +24,13 @@ Feature: delete folder contents
...
       | /FOLDER/SUBFOLDER/              |
-      | /FOLDER/fileToDelete.txt             |
+      | /FOLDER/fileToDelete.txt        |
       | /FOLDER/SUBFOLDER/testfile0.txt |
     Examples:
       | dav_version |
       | old         |
       | new         |
+
+    Examples:
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFolderContents.feature:34]:
```suggestion
 @skipOnOcV10 @personalSpace
    Examples:
```

File: tests/TestHelpers/WebDavHelper.php [hunk 2]
```diff
@@ -513,6 +513,7 @@ public static function makeDavRequest(
 				}
 			}
 		}
+
```
Comment (amrita-shrestha) [inline tests/TestHelpers/WebDavHelper.php:516]: unnecessary line added

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 3]
```diff
@@ -19,6 +19,11 @@ Feature: delete file
       | old         |
       | new         |
 
+    @skipOnOcV10 @personalSpace
+    Examples:
+      | dav_version |
+      | spaces      |
+
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature:26]:
```suggestion


```

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 4]
```diff
@@ -33,6 +38,11 @@ Feature: delete file
...
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature:45]:
```suggestion


```

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 5]
```diff
@@ -55,6 +67,17 @@ Feature: delete file
...
+    Examples:
+      | dav_version    | folder_name   | file_name   |
+      | spaces         | /upload.      | abc.        |
+      | spaces         | /upload...1.. | abc...txt.. |
+      | spaces         | /upload.1     | abc.txt     |
+      | spaces         | /upload.      | abc .       |
+      | spaces         | /...          | ...         |
+      | spaces         | /..upload     | abc         |
+      | spaces         | /..upload     | ...abc      |
+
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature:80]:
```suggestion


```

File: tests/TestHelpers/WebDavHelper.php (hunk 2 above)
Comment (saw-jan) [inline tests/TestHelpers/WebDavHelper.php:516]: we may not need this change or do we?

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 6]
```diff
@@ -70,6 +93,13 @@ Feature: delete file
       | new         | ",,,.txt"      |
       | new         | ",,,.,"        |
 
+    @skipOnOcV10 @personalSpace
+    Examples:
+      | dav_version    | filename       |
+      | spaces         | "sample,1.txt" |
+      | spaces         | ",,,.txt"      |
+      | spaces         | ",,,.,"        |
+
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature:102]:
```suggestion


```

File: tests/acceptance/features/apiWebdavOperations/deleteFolder.feature [hunk 7]
```diff
@@ -97,19 +112,30 @@ Feature: delete folder
     When user "Alice" deletes folder "<folder_name>" using the WebDAV API
     Then the HTTP status code should be "204"
     And as "Alice" folder "<folder_name>" should not exist
+#    Examples:
+#      | dav_version | folder_name |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFolder.feature:116]: uncomment examples
```suggestion
     | dav_version | folder_name |
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 8]
```diff
@@ -283,24 +283,20 @@ public function usingSpacesDavPath():void {
...
 			);
-			$spaceDavPath = "dav/spaces/users/" . $spaceId . '/';
-			$path = $this->getBasePath() . "/" .
-				$spaceDavPath;
-		} else {
-			$path = $this->getBasePath() . "/" .
-				WebDavHelper::getDavPath($user, $this->getDavPathVersion());
 		}
+		$path = $this->getBasePath() . "/" .
+			WebDavHelper::getDavPath($user, $this->getDavPathVersion(), "files", $spaceId);
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/WebDav.php:299]: Why this changes is done? Can $path for dav version old and new be set correct by this?

File: tests/acceptance/features/apiWebdavOperations/deleteFolder.feature [hunk 9]
```diff
@@ -97,19 +112,30 @@ Feature: delete folder
...
+#      | old         | /fo.xyz     |
+#      | old         | /fo.exe     |
+#      | new         | /fo.        |
+#      | new         | /fo.1       |
+#      | new         | /fo...1..   |
+#      | new         | /...        |
+#      | new         | /..fo       |
+#      | new         | /fo.xyz     |
+#      | new         | /fo.exe     |
+
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/deleteFolder.feature:131]:
```suggestion
    Examples:
      | dav_version | folder_name |
      | old         | /fo.        |
      | old         | /fo.1       |
      | old         | /fo...1..   |
      | old         | /...        |
      | old         | /..fo       |
      | old         | /fo.xyz     |
      | old         | /fo.exe     |
      | new         | /fo.        |
      | new         | /fo.1       |
      | new         | /fo...1..   |
      | new         | /...        |
      | new         | /..fo       |
      | new         | /fo.xyz     |
      | new         | /fo.exe     |

```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 10]
```diff
@@ -283,24 +283,20 @@ public function usingSpacesDavPath():void {
 	 * @param string $user
 	 *
 	 * @return string
+	 * @throws GuzzleException
 	 */
 	public function getFullDavFilesPath(string $user):string {
 		$spaceId = null;
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/WebDav.php:289]: Please delete this line also

**Comment by kiranparajuli589** [comment]
sorry the [ci build](https://drone.owncloud.com/owncloud/core/34498) failed at `install-core` in several suites, build restarted

File: tests/TestHelpers/WebDavHelper.php [hunk 11]
```diff
@@ -398,19 +398,21 @@ public static function getPersonalSpaceIdForUser(string $baseUrl, string $user,
...
-				$personalSpaceId = $spaces->id;
-				break;
+		try {
+			foreach ($json->value as $spaces) {
+				if ($spaces->driveType === "personal") {
+					$personalSpaceId = $spaces->id;
+					break;
+				}
 			}
+		} finally {
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:408]: use `catch`

File: tests/acceptance/features/apiWebdavOperations/deleteFolder.feature [hunk 12]
```diff
@@ -112,4 +128,15 @@ Feature: delete folder
       | new         | /...        |
       | new         | /..fo       |
       | new         | /fo.xyz     |
-      | old         | /fo.exe     |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavOperations/deleteFolder.feature:115]: good catch! :fishing_pole_and_fish: :+1:

File: tests/acceptance/features/apiWebdavOperations/downloadFile.feature [hunk 13]
```diff
@@ -235,3 +317,10 @@ Feature: download file
       | dav_version |
       | old         |
       | new         |
+
+    @personalSpace
+    Examples:
+      | dav_version |
+      | spaces      |
+
+
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavOperations/downloadFile.feature:326]: remove this last line

File: tests/acceptance/features/apiWebdavOperations/listFiles.feature [hunk 14]
```diff
@@ -59,6 +59,7 @@ Feature: list files
       | simple-folder/                                            |
       | simple-folder/textfile0.txt                               |
       | simple-folder/welcome.txt                                 |
+      | simple-folder/simple-empty-folder/                        |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavOperations/listFiles.feature:62]: :tophat:

File: tests/acceptance/features/apiWebdavOperations/refuseAccess.feature [hunk 15]
```diff
@@ -18,8 +19,8 @@ Feature: refuse access
       | WWW-Authenticate | Basic realm="%productname%", charset="UTF-8" |
     Examples:
       | dav_version |
-      | old         |
       | new         |
+      | old         |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavOperations/refuseAccess.feature:23]: revert this please

**Review by kiranparajuli589 — DISMISSED** [review]

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 8 above)
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/WebDav.php:299]: I think this changes affect 
apiFavorites/favorites.feature:117
apiFavorites/favorites.feature:143
apiFavorites/favorites.feature:268

Comment (SagarGi) [reply tests/acceptance/features/bootstrap/WebDav.php:299]: ok i will check it.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 8 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/bootstrap/WebDav.php]: i had created issue for that bug https://github.com/owncloud/ocis/issues/3111

**Comment by phil-davis** [comment]
Issue #39549 `webUIUpload/upload.feature:131` failed. Unrelated to this PR. I restarted CI.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@SagarGi I just merged PR #39745 that @amrita-shrestha made last week. It had similar changes to `tests/acceptance/features/bootstrap/WebDav.php` `getFullDavFilesPath` but not quite the same.

Please rebase this PR and sort out if there are any more changes needed to `getFullDavFilesPath`, or if her changes are OK.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**